### PR TITLE
Use `AirflowProviderDeprecationWarning` in providers

### PR DESCRIPTION
**Pull request by eladkal**
Now that Providers minimum Airflow version is 2.4.0 we can replace `DeprecationWarning` with `AirflowProviderDeprecationWarning`

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/sqlite/operators/sqlite.py [hunk 1]
```diff
@@ -49,6 +50,6 @@ def __init__(self, *, sqlite_conn_id: str = "sqlite_default", **kwargs) -> None:
         warnings.warn(
             """This class is deprecated.
             Please use `airflow.providers.common.sql.operators.sql.SQLExecuteQueryOperator`.""",
-            DeprecationWarning,
+            AirflowProviderDeprecationWarning,
```
Comment (eladkal) [inline airflow/providers/sqlite/operators/sqlite.py:53]: Not sure about this one.
sqlite provider doesn't have min Airflow version (as it's one of the pre installed providers)
Nothing blocks users from updating this provider independently with older version of Airflow which may result in import error.

I wonder if we have a way to prevent PIP from allowing it?

Comment (potiuk) [reply airflow/providers/sqlite/operators/sqlite.py:53]: Theorethically we could add dependency on airflow for pip (we used to have them) but it breaks other tools that detect it as cycles  (pip is fine), what we could do is that we could do similar thing as some other packages do - check it at runtime when it is used and raise a warning or error. This way if someone installs a new version of sqlite on old airflow version it would refuse to start. 

It won't prevent it from being upgraded, but at least we can raise a clear error message in this case and ask the user to upgrade airflow if they want to update the provider to a newer version. 

I think we should do it anyway for all preinstalled packages - WDYT @eladkal ?

Comment (eladkal) [reply airflow/providers/sqlite/operators/sqlite.py:53]: yeah that is a good idea to do anyway!

My question here is to raise the two options:
1. Since our policy states that all providers are 2.4 compatible leave this as is and if users will face it then they will have to downgrade - Not user friendly and I hope to avoid this!
2. Add try/catch to support both imports thus being compatible with older versions of Airflow. This is kinda against our policy but this prevents the potential issue.

Comment (potiuk) [reply airflow/providers/sqlite/operators/sqlite.py:53]: I'd be for raising hard error. One of the reasons for having min-airflow version is to be able to AVOID this kind of hacks - even if theorethically the provider could work with older airflow. This is also a way to have our users more-or-less-gently pushed to upgrade airflow.

Comment (eladkal) [reply airflow/providers/sqlite/operators/sqlite.py:53]: that is reasonable.
Can this be done in the provider `__init.py__` ?
Something like:
```
    from airflow.version import version
    if Version(version) < Version("2.4"):
       raise ....
```

**Review by vincbeck — APPROVED** [review]

File: airflow/providers/sqlite/operators/sqlite.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/sqlite/operators/sqlite.py:53]: Yes. That was very close to what I implemented in https://github.com/apache/airflow/pull/30994

**Review by pankajastro — COMMENTED** [review]
does it makes sense to add a pre-commit to enforce contributor to use `AirflowProviderDeprecationWarning` and not `DeprecationWarning` in providers I feel that it would be really nice if we know in which version the deprecation is added?

File: airflow/providers/sqlite/operators/sqlite.py (hunk 1 above)
Comment (eladkal) [reply airflow/providers/sqlite/operators/sqlite.py:53]: cool. Rebased after https://github.com/apache/airflow/pull/30994
lets see

File: airflow/providers/amazon/aws/operators/eks.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 from botocore.exceptions import ClientError, WaiterError
 
 from airflow import AirflowException
+from airflow.exceptions import AirflowProviderDeprecationWarning
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/operators/eks.py]:
```suggestion
from airflow.exceptions import AirflowException, AirflowProviderDeprecationWarning
```

Drive-by improvements.

File: airflow/providers/dbt/cloud/sensors/dbt.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 from typing import TYPE_CHECKING, Any
 
 from airflow import AirflowException
+from airflow.exceptions import AirflowProviderDeprecationWarning
```
Comment (uranusjr) [inline airflow/providers/dbt/cloud/sensors/dbt.py]:
```suggestion
from airflow.exceptions import AirflowException, AirflowProviderDeprecationWarning
```

File: airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py [hunk 4]
```diff
@@ -65,6 +65,7 @@
 from google.cloud.aiplatform_v1.types import TrainingPipeline
 
 from airflow import AirflowException
+from airflow.exceptions import AirflowProviderDeprecationWarning
```
Comment (uranusjr) [inline airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py]:
```suggestion
from airflow.exceptions import AirflowException, AirflowProviderDeprecationWarning
```

File: airflow/providers/google/cloud/operators/dataflow.py [hunk 5]
```diff
@@ -28,6 +28,7 @@
 
 from airflow import AirflowException
 from airflow.compat.functools import cached_property
+from airflow.exceptions import AirflowProviderDeprecationWarning
```
Comment (uranusjr) [inline airflow/providers/google/cloud/operators/dataflow.py:31]:
```suggestion
from airflow.compat.functools import cached_property
from airflow.exceptions import AirflowException, AirflowProviderDeprecationWarning
```

File: tests/providers/amazon/aws/transfers/test_dynamodb_to_s3.py [hunk 6]
```diff
@@ -25,6 +25,7 @@
 import pytest
 
 from airflow import DAG
+from airflow.exceptions import AirflowProviderDeprecationWarning
 from airflow.models import DagRun, TaskInstance
```
Comment (uranusjr) [inline tests/providers/amazon/aws/transfers/test_dynamodb_to_s3.py:29]:
```suggestion
from airflow.exceptions import AirflowProviderDeprecationWarning
from airflow.models import DAG, DagRun, TaskInstance
```

**Review by uranusjr — APPROVED** [review]
There are some more but GitHub doesn’t allow me to add them. Those are trivial anyway and don’t strictly need to be added here.
